e2e_playwright/st_text_area_test.py

```diff
@@ -0,0 +1,218 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+from playwright.sync_api import Page, expect
+
+from e2e_playwright.conftest import ImageCompareFunction
+
+
+def test_text_area_widget_rendering(
+    themed_app: Page, assert_snapshot: ImageCompareFunction
+):
+    """Test that the st.text_area widgets are correctly rendered via screenshot matching."""
+    text_area_widgets = themed_app.get_by_test_id("stTextArea")
+    expect(text_area_widgets).to_have_count(11)
+
+    assert_snapshot(text_area_widgets.nth(0), name="st_text_area-default")
+    assert_snapshot(text_area_widgets.nth(1), name="st_text_area-value_some_text")
+    assert_snapshot(text_area_widgets.nth(2), name="st_text_area-value_1234")
+    assert_snapshot(text_area_widgets.nth(3), name="st_text_area-value_None")
+    assert_snapshot(text_area_widgets.nth(4), name="st_text_area-placeholder")
+    assert_snapshot(text_area_widgets.nth(5), name="st_text_area-disabled")
+    assert_snapshot(text_area_widgets.nth(6), name="st_text_area-hidden_label")
+    assert_snapshot(text_area_widgets.nth(7), name="st_text_area-collapsed_label")
+    assert_snapshot(text_area_widgets.nth(8), name="st_text_area-callback_help")
+    assert_snapshot(text_area_widgets.nth(9), name="st_text_area-max_chars_5")
+    assert_snapshot(text_area_widgets.nth(10), name="st_text_area-height_250")
+
+
+def test_text_area_has_correct_initial_values(app: Page):
+    """Test that st.text_area has the correct initial values."""
+    markdown_elements = app.get_by_test_id("stMarkdown")
+    expect(markdown_elements).to_have_count(12)
+
+    expected = [
+        "value 1: ",
+        "value 2: some text",
+        "value 3: 1234",
+        "value 4: None",
+        "value 5: ",
+        "value 6: default text",
+        "value 7: default text",
+        "value 8: default text",
+        "value 9: ",
+        "text area changed: False",
+        "value 10: 1234",
+        "value 11: default text",
+    ]
+
+    for markdown_element, expected_text in zip(markdown_elements.all(), expected):
+        expect(markdown_element).to_have_text(expected_text, use_inner_text=True)
+
+
+def test_text_area_shows_instructions_when_dirty(
+    app: Page, assert_snapshot: ImageCompareFunction
+):
+    """Test that st.text_area shows the instructions correctly when dirty."""
+    text_area = app.get_by_test_id("stTextArea").nth(9)
+
+    text_area_field = text_area.locator("textarea").first
+    text_area_field.fill("123")
+
+    assert_snapshot(text_area, name="st_text_area-input_instructions")
+
+
+def test_text_area_limits_input_via_max_chars(app: Page):
+    """Test that st.text_area correctly limits the number of characters via max_chars."""
+    first_text_area_field = (
```

Comment:
nit: the variable name `first_text_area_field` is a bit confusing here since it's actually the 9th

Reply:
whoops, that wasn't intended